Open Judge System/OJS.Services/Business/ExamGroups/ExamGroupsBusinessService.cs

```diff
@@ -0,0 +1,94 @@
+﻿namespace OJS.Services.Business.ExamGroups
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Web;
+
+    using OJS.Common;
+    using OJS.Services.Common.HttpRequester;
+    using OJS.Services.Common.HttpRequester.Models.Users;
+    using OJS.Services.Data.ExamGroups;
+    using OJS.Services.Data.Users;
+
+    public class ExamGroupsBusinessService : IExamGroupsBusinessService
+    {
+        // TODO: Add to Resource
+        private const string ExamGroupCannotBeNullMessage = "Exam group cannot be null";
+
+        private readonly IExamGroupsDataService examGroupsData;
+        private readonly IUsersDataService usersData;
+        private readonly IHttpRequesterService httpRequester;
+        private readonly string sulsPlatformBaseUrl;
+        private readonly string apiKey;
+
+        public ExamGroupsBusinessService(
+            IExamGroupsDataService examGroupsData,
+            IUsersDataService usersData,
+            IHttpRequesterService httpRequester,
+            string sulsPlatformBaseUrl,
+            string apiKey)
+        {
+            this.examGroupsData = examGroupsData;
+            this.usersData = usersData;
+            this.httpRequester = httpRequester;
+            this.sulsPlatformBaseUrl = sulsPlatformBaseUrl;
+            this.apiKey = apiKey;
+        }
+
+        public void AddUsersByIdAndUserIds(int id, IEnumerable<string> userIds)
+        {
+            var examGroup = this.examGroupsData.GetById(id);
+
+            if (examGroup == null)
+            {
+                throw new ArgumentNullException(nameof(examGroup), ExamGroupCannotBeNullMessage);
+            }
+
+            foreach (var userId in userIds)
+            {
+                var user = this.usersData.GetByIdIncludingDeleted(userId);
+
+                if (user == null)
```

Comment:
Extract the logic for getting user from platform into a separate method and use in a FireAndForget background job